### PR TITLE
Workaround Travis-side CI issues

### PR DESCRIPTION
**Pull request by luke-jr**
Currently Travis's wget fails fetching qrencode:

```
Fetching qrencode...
ERROR: no certificate subject alternative name matches
    requested host name `fukuchi.org'.
To connect to fukuchi.org insecurely, use `--no-check-certificate'.
OpenSSL: error:14077438:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert internal error
Unable to establish SSL connection.
make: *** [/home/travis/build/luke-jr/bitcoin/depends/sources/download-stamps/.stamp_fetched-qrencode-qrencode-3.4.4.tar.bz2.hash] Error 4
```

### CODE REVIEW

**Comment by laanwj** [comment]
What wonders me most is why it won't try the fallback URL

**Comment by luke-jr** [comment]
Not sure, but even if it did, it would fail. Note the second commit I had to add for the OSX SDK on bitcoincore.org :/

**Comment by laanwj** [comment]
Why is the MacOSX SDK download broken too?

I don't think I understand what is the issue here. Is that site irrevocably broken, or is there some certificate issue? (e.g. missing ca-certificates package?) But in that case you'd expect any tool to fail.

Switching to another tool is possible but I'd like to know what is wrong first. It seems like a heavy-handed solution.

@theuni any idea?

**Comment by luke-jr** [comment]
Seems to be a wget bug in the ancient 12.02 Ubuntu that Travis uses. But I could be wrong.

**Comment by laanwj** [comment]
Bleh. We need 14.04 support for other reasons too (e.g. c++11).

**Comment by luke-jr** [comment]
It looks like 14.04 is possible, but... only be losing IPv6 localhost support [again]. :/

**Comment by dcousens** [comment]
concept ACK

**Comment by luke-jr** [comment]
Note: curl needs -L to follow redirects.

**Comment by luke-jr** [comment]
This fixes the Travis-side CI issues: https://travis-ci.org/luke-jr/bitcoin/builds/108146026

**Comment by btcdrak** [comment]
utACK, agree on using curl

File: depends/builders/linux.mk [hunk 1]
```diff
@@ -1,2 +1,2 @@
 build_linux_SHA256SUM = sha256sum
-build_linux_DOWNLOAD = wget --timeout=$(DOWNLOAD_CONNECT_TIMEOUT) --tries=$(DOWNLOAD_RETRIES) -nv -O
+build_linux_DOWNLOAD = curl --location --fail --connect-timeout $(DOWNLOAD_CONNECT_TIMEOUT) --retry $(DOWNLOAD_RETRIES) -L -o
```
Comment (MarcoFalke) [inline depends/builders/linux.mk]: Is it required to set `--location` and `-L`?

**Comment by MarcoFalke** [comment]
utACK  c01f08d

**Pull request merged**